### PR TITLE
Expose driver base classes via API

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Synopsis**

#518 exposed component-driver classes via the `uwtools` API, but externally-developed coupled drivers will also need to access base driver classes (see #521) to inherit from and extend. This PR provides access.

**Type**

- [x] Enhancement (adds new functionality)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/api/driver.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+def _add_classes():
+    m = importlib.import_module("uwtools.drivers.driver")
+    for classname in _CLASSNAMES:
+        setattr(sys.modules[__name__], classname, getattr(m, classname))
+        __all__.append(classname)
+
+
+__all__: list[str] = []
+_add_classes()
```
Comment (maddenp-noaa) [inline src/uwtools/api/driver.py:28]: This module defines the driver base classes to expose to the API and adds them to this module.

File: src/uwtools/drivers/ww3.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 class WaveWatchIII(AssetsCycleBased):
     """
-    A library driver for ww3.
+    An assets driver for ww3.
```
Comment (maddenp-noaa) [inline src/uwtools/drivers/ww3.py:17]: For consistency with docstrings in other modules and `uwtools` terminology.

File: src/uwtools/tests/api/test_driver.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+from uwtools.drivers import driver as driver_lib
+
+
+@mark.parametrize("classname", driver_api._CLASSNAMES)
+def test_driver(classname):
+    assert getattr(driver_api, classname) is getattr(driver_lib, classname)
+
+
+def test_public_attributes():
+    # Check that the module is not accidentally exposing unexpected public atttributes. Ignore
```
Comment (elcarpenterNOAA) [inline src/uwtools/tests/api/test_driver.py:17]:
```suggestion
    # Check that the module is not accidentally exposing unexpected public attributes. Ignore
```

**Review by NaureenBharwaniNOAA — APPROVED** [review]

**Review by elcarpenterNOAA — APPROVED** [review]

File: src/uwtools/api/driver.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+_CLASSNAMES = [
+    "Assets",
+    "AssetsCycleBased",
+    "AssetsCycleAndLeadtimeBased",
+    "AssetsTimeInvariant",
+    "Driver",
+    "DriverCycleBased",
+    "DriverCycleAndLeadtimeBased",
+    "DriverTimeInvariant",
+]
```
Comment (christinaholtNOAA) [inline src/uwtools/api/driver.py:17]: It's pretty nifty to have a place to see what all our drivers' names are. I know there are other ways to get the info, but this is nice.

**Review by christinaholtNOAA — APPROVED** [review]
LGTM!

**Pull request merged**